src/systemTest/java/com/flightstats/hub/testcase/WebhookTest.java

```diff
@@ -0,0 +1,221 @@
+import com.flightstats.hub.client.ChannelResourceClient;
+import com.flightstats.hub.client.WebhookResourceClient;
+import com.flightstats.hub.model.Channel;
+import com.flightstats.hub.model.ChannelItem;
+import com.flightstats.hub.model.ContentKey;
+import com.flightstats.hub.model.Webhook;
+import com.flightstats.hub.model.WebhookErrors;
+import lombok.SneakyThrows;
+import org.junit.After;
+import org.slf4j.Logger;
+import retrofit2.Call;
+import retrofit2.Response;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static com.flightstats.hub.model.ChannelContentStorageType.SINGLE;
+import static javax.ws.rs.core.Response.Status.CREATED;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+
+public abstract class WebhookTest extends BaseTest {
+    private static final String EMPTY_STRING = "";
+    private Logger log;
+    protected ChannelItemResourceClient channelItemResourceClient;
+    protected ChannelResourceClient channelResourceClient;
+    protected WebhookResourceClient webhookResourceClient;
+    protected CallbackResourceClient callbackResourceClient;
+    @Inject
+    protected CallbackServer callbackServer;
+    @Inject
+    protected CallbackResource callbackResource;
+    protected String channelName;
+    protected String webhookName;
+
+    public void setup() {
+        super.setup();
+        this.log = getLog();
+        this.channelItemResourceClient = getHubClient(ChannelItemResourceClient.class);
+        this.channelResourceClient = getHubClient(ChannelResourceClient.class);
+        this.webhookResourceClient = getHubClient(WebhookResourceClient.class);
+
+        this.callbackResourceClient = getCallbackClient(CallbackResourceClient.class);
+        this.callbackServer.start(super.injector);
+
+        this.channelName = generateRandomString();
+        this.webhookName = generateRandomString();
+    }
+
+    protected abstract Logger getLog();
+
+    @SneakyThrows
+    protected void logCurrentCallbackErrorsFromHub() {
+        log.info("Webhook name {} ", webhookName);
+        WebhookErrors body = getCallbackErrorsFromHub().body();
+        log.info("Call back errors for webhook {} {} ", webhookName, body);
+    }
+
+    @SneakyThrows
+    protected Response<WebhookErrors> getCallbackErrorsFromHub() {
+        Call<WebhookErrors> call = webhookResourceClient.getError(webhookName);
+        Response<WebhookErrors> response = call.execute();
+
+        WebhookErrors body = response.body();
+        assertEquals(OK.getStatusCode(), response.code());
```

Comment:
If these assertions are gut-checks, I think it's fine to keep these here. If not, I think moving the assertions into a different method that calls out that they assert something may make sense.

Reply:
ok. This is coz the response body is logged in the next statement. If for some reason the service call fails, it will result in NPE. Assertion is way of logging the response code from the service call in case of error.